src/Core/Content/Seo/SeoUrlGenerator.php

```diff
@@ -116,11 +132,15 @@ private function generateUrls(SeoUrlRouteInterface $seoUrlRoute, SeoUrlRouteConf
         }
     }
 
-    private function getSeoPathInfo(SeoUrlMapping $mapping, SeoUrlRouteConfig $config): ?string
+    private function getSeoPathInfo(Environment $twig, SeoUrlMapping $mapping, SeoUrlRouteConfig $config): ?string
     {
         try {
-            return trim($this->twig->render('template', $mapping->getSeoPathInfoContext()));
-        } catch (\Throwable $error) {
+            return trim($twig->render('template', $mapping->getSeoPathInfoContext()));
```

Comment:
does the twig caching still work?
are there any performance differences between using one instance and creating always one?

Reply:
The factory disables the twig cache anyway.

Reply:
Yes there is. First, you don't create the first twig environment in the container anymore. Second, you create an environment for each indexing call so either in a web requests which is similar to have it in the container or per indexing message in the message worker which is neglectable in impact IMO.